Zend/zend_compile.c

```diff
@@ -3054,6 +3056,19 @@ static zend_op *zend_delayed_compile_prop(znode *result, zend_ast *ast, uint32_t
 			}
 			zend_emit_jmp_null(&obj_node, type);
 		}
+
+		/* ZEND_AST_CONST can result in IS_TMP_VAR and IS_CONST which is uncommon for accessing
+		 * properties. ASSIGN_OBJ and FETCH_OBJ_[R]W are missing handlers for CONST|TMP op1, we work
+		 * around this by converting the TMP to a VAR, and by assigning the CONST to a temporary VAR. */
+		if ((type == BP_VAR_W || type == BP_VAR_RW)) {
+			if (obj_node.op_type == IS_TMP_VAR) {
+				ZEND_ASSERT(obj_ast->kind == ZEND_AST_CONST);
+				obj_node.op_type = IS_VAR;
+			} else if (obj_node.op_type == IS_CONST) {
+				ZEND_ASSERT(obj_ast->kind == ZEND_AST_CONST);
+				zend_emit_op(&obj_node, ZEND_QM_ASSIGN, &obj_node, NULL);
```

Comment:
Do we need this code just to throw exception at run-time?

Reply:
Indeed...